lib/request.js

```diff
@@ -1,254 +0,0 @@
-      }
-      if (context.request.url.port) {
-        context.request.url.port = truncate(String(context.request.url.port), config.INTAKE_STRING_MAX_SIZE)
-      }
-      if (context.request.url.pathname) {
-        context.request.url.pathname = truncate(String(context.request.url.pathname), config.INTAKE_STRING_MAX_SIZE)
-      }
-      if (context.request.url.search) {
-        context.request.url.search = truncate(String(context.request.url.search), config.INTAKE_STRING_MAX_SIZE)
-      }
-      if (context.request.url.hash) {
-        context.request.url.hash = truncate(String(context.request.url.hash), config.INTAKE_STRING_MAX_SIZE)
-      }
-      if (context.request.url.raw) {
-        context.request.url.raw = truncate(String(context.request.url.raw), config.INTAKE_STRING_MAX_SIZE)
-      }
-      if (context.request.url.full) {
-        context.request.url.full = truncate(String(context.request.url.full), config.INTAKE_STRING_MAX_SIZE)
-      }
-    }
-  }
-  if (context.user) {
-    if (context.user.id) {
-      context.user.id = truncate(String(context.user.id), config.INTAKE_STRING_MAX_SIZE)
-    }
-    if (context.user.email) {
-      context.user.email = truncate(String(context.user.email), config.INTAKE_STRING_MAX_SIZE)
-    }
-    if (context.user.username) {
-      context.user.username = truncate(String(context.user.username), config.INTAKE_STRING_MAX_SIZE)
-    }
-  }
-}
-
-function truncFrames (frames) {
-  frames.forEach(function (frame, i) {
-    if (frame.pre_context) frame.pre_context = truncEach(frame.pre_context, 1000)
-    if (frame.context_line) frame.context_line = truncate(String(frame.context_line), 1000)
-    if (frame.post_context) frame.post_context = truncEach(frame.post_context, 1000)
-  })
-
-  return frames
-}
-
-function truncEach (arr, len) {
-  return arr.map(function (str) {
-    return truncate(String(str), len)
-  })
-}
-
-// Used only for debugging data sent to the intake API
-function capturePayload (endpoint, payload) {
-  var dumpfile = path.join(os.tmpdir(), 'elastic-apm-node-' + endpoint + '-' + Date.now() + '.json')
-  fs.writeFile(dumpfile, JSON.stringify(payload), function (err) {
-    if (err) console.log('could not capture intake payload: %s', err.message)
-    else console.log('intake payload captured: %s', dumpfile)
-  })
-}
-
-function envelope (agent) {
-  var payload = {
-    service: {
-      name: agent._conf.serviceName,
-      runtime: {
-        name: process.release.name,
-        version: process.version
-      },
-      language: {
-        name: 'javascript'
-      },
-      agent: {
-        name: 'nodejs',
-        version: AGENT_VERSION
-      }
-    },
-    process: {
-      pid: process.pid,
-      ppid: process.ppid,
-      title: truncate(String(process.title), config.INTAKE_STRING_MAX_SIZE),
-      argv: process.argv
-    },
-    system: {
-      hostname: agent._conf.hostname,
-      architecture: process.arch,
-      platform: process.platform
-    }
-  }
```

Comment:
It seems like process, system and runtime stuff no longer exists? I don't see it in the http client PR either. Is that correct?

Reply:
Yeah, I can see it's a bit confusing given the current state of the all the in-progress dependencies.

As you can see here, this PR depends on a custom branch of the http client called `v2-1` that currently lives on my fork: https://github.com/elastic/apm-agent-nodejs/pull/465/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R74

In that branch the metadata responsibilities have been moved to the http client: https://github.com/watson/apm-nodejs-http-client/blob/4f02d7eda173b738a309fdca2e3464a1bbae5652/index.js#L293-L332

But there's no PR for this branch yet as I don't want to make the current http client PR too complicated, so once the the current PR is merged, I'll open a new one for the `v2-1` branch 😅

Reply:
Just merge the first http client PR and created the next based on my `v2-1` branch: https://github.com/elastic/apm-nodejs-http-client/pull/6 - soon it should hopefully be less confusing

Reply:
Ah. Gotcha. I was trying to match up the two PRs to make sure the functionality lined up between the two. 😅